backend/battle/forms.py

```diff
@@ -20,13 +27,43 @@ def __init__(self, *args, **kwargs):
         super(BattleForm, self).__init__(*args, **kwargs)
         self.fields['creator'].widget = forms.HiddenInput()
 
+    def clean_opponent(self):
+        opponent_email = self.cleaned_data["opponent"]
+        try:
+            opponent = User.objects.get(email=opponent_email)
+        except User.DoesNotExist:
+            self.is_guest = True
+            opponent = User.objects.create(email=opponent_email)
+            random_password = get_random_string(length=64)
+            opponent.set_password(random_password)
+            opponent.save()
+        return opponent
+
     def clean(self):
         cleaned_data = super().clean()
         if 'creator' not in cleaned_data:
             raise forms.ValidationError("ERROR: You need to be logged.")
 
         if cleaned_data['opponent'] == cleaned_data['creator']:
             raise forms.ValidationError("ERROR: You can't challenge yourself.")
+        return cleaned_data
+
+    def save(self, commit=True):
+        cleaned_data = self.clean()
+        instance = super().save()
+        opponent = cleaned_data["opponent"]
+        if self.is_guest:
+            invite_form = PasswordResetForm(data={"email": opponent.email})
+            invite_form.is_valid()
+            invite_form.save(
+                self, subject_template_name='registration/password_reset_subject.txt',
```

Comment:
You should use your own templates for subject and email. And they should have different file names and contents.
Make the contents reflect the behavior here, which is an invite, not a password reset.

Reply:
Got it, thanks!